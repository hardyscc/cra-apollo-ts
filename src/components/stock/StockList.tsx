import * as React from 'react'
import { Query } from 'react-apollo'
import { Table } from 'antd'
import { gql } from 'apollo-boost'
import { GetStocksQuery } from '../../__generated__/types'

class StocksQuery extends Query<GetStocksQuery> {}

export const STOCKS_QUERY = gql`
  query GetStocks {
    stocks(type: EQTY, first: 100, skip: 0) {
      code
      name
      lotSize
      status
    }
  }
`

export const StockList = () => (
  <StocksQuery query={STOCKS_QUERY}>
    {({ loading, error, data }) => {
      if (loading) return <div>Loading...</div>
      if (error) return <div>Error :(</div>

      return (
        <div data-testid="stock-list">
          <Table dataSource={data && data.stocks} size="small">
            <Table.Column title="Code" dataIndex="code" key="code" />
            <Table.Column title="Name" dataIndex="name" key="name" />
            <Table.Column title="Lot Size" dataIndex="lotSize" key="lotSize" />
            <Table.Column title="Status" dataIndex="status" key="status" />
          </Table>
        </div>
      )
    }}
  </StocksQuery>
)
