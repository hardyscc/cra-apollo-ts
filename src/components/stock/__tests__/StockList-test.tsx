import * as React from 'react'
import { render, wait } from 'react-testing-library'
import { MockedProvider } from 'react-apollo/test-utils'
import { StockList, STOCKS_QUERY } from '../StockList'

const stocks = [
  {
    __typename: 'Stock',
    code: '0001.HK',
    lotSize: 50,
    name: '長和',
    status: 'Active'
  },
  {
    __typename: 'Stock',
    code: '0002.HK',
    lotSize: 100,
    name: '中電控股',
    status: 'Active'
  }
]
const request = { query: STOCKS_QUERY }
const result = { data: { stocks } }
const error = new Error('something go wrong')

describe('App', () => {
  it('render', async () => {
    const { container, getByTestId } = render(
      <MockedProvider mocks={[{ request, result }]}>
        <StockList />
      </MockedProvider>
    )
    await wait(() => getByTestId('stock-list'))
    expect(container).toMatchSnapshot()
  })

  it('error', async () => {
    const { container, getByText } = render(
      <MockedProvider mocks={[{ request, error }]}>
        <StockList />
      </MockedProvider>
    )
    await wait(() => getByText('Error!'))
    expect(container).toMatchSnapshot()
  })
})
