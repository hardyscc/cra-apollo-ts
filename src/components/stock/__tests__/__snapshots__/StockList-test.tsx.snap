// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App error 1`] = `
<div>
  <div>
    Error :(
  </div>
</div>
`;

exports[`App render 1`] = `
<div>
  <div
    data-testid="stock-list"
  >
    <div
      class="ant-table-wrapper"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table ant-table-small ant-table-scroll-position-left"
          >
            <div
              class="ant-table-content"
            >
              <div
                class="ant-table-body"
              >
                <table
                  class=""
                >
                  <colgroup>
                    <col />
                    <col />
                    <col />
                    <col />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class=""
                      >
                        <span>
                          Code
                        </span>
                      </th>
                      <th
                        class=""
                      >
                        <span>
                          Name
                        </span>
                      </th>
                      <th
                        class=""
                      >
                        <span>
                          Lot Size
                        </span>
                      </th>
                      <th
                        class=""
                      >
                        <span>
                          Status
                        </span>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row  ant-table-row-level-0"
                    >
                      <td
                        class=""
                      >
                        <span
                          class="ant-table-row-indent indent-level-0"
                          style="padding-left: 0px;"
                        />
                        0001.HK
                      </td>
                      <td
                        class=""
                      >
                        長和
                      </td>
                      <td
                        class=""
                      >
                        50
                      </td>
                      <td
                        class=""
                      >
                        Active
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row  ant-table-row-level-0"
                    >
                      <td
                        class=""
                      >
                        <span
                          class="ant-table-row-indent indent-level-0"
                          style="padding-left: 0px;"
                        />
                        0002.HK
                      </td>
                      <td
                        class=""
                      >
                        中電控股
                      </td>
                      <td
                        class=""
                      >
                        100
                      </td>
                      <td
                        class=""
                      >
                        Active
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <ul
            class="ant-pagination ant-table-pagination mini"
            unselectable="unselectable"
          >
            <li
              aria-disabled="true"
              class="ant-pagination-disabled ant-pagination-prev"
              title="Previous Page"
            >
              <a
                class="ant-pagination-item-link"
              />
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
              tabindex="0"
              title="1"
            >
              <a>
                1
              </a>
            </li>
            <li
              aria-disabled="true"
              class="ant-pagination-disabled ant-pagination-next"
              title="Next Page"
            >
              <a
                class="ant-pagination-item-link"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
